drivers/sensor/htu21d/htu21d.c

```diff
@@ -0,0 +1,164 @@
+/* HTU21D temperature & humidity sensor with i2c
+ *
+ * Copyright (c) 2022 Linumiz
+ * Author: Arunmani <arunmani@linumiz.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT te_htu21d
+#include <zephyr.h>
+#include <stdio.h>
+#include "htu21d.h"
+#include <logging/log.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <drivers/sensor.h> 
+#include <drivers/i2c.h>
+#define SIXTEEN_BIT	65536.0     
+#define TEMPERATURE_COEFF_MUL	175.72
+#define TEMPERATURE_COEFF_ADD	-46.85
+#define HUMIDITY_COEFF_MUL	125.0
+#define HUMIDITY_COEFF_ADD	-6.0
+#define DECIMAL		10000
+#define MICRO	100
+
+LOG_MODULE_REGISTER(htu21d, CONFIG_SENSOR_LOG_LEVEL);
+
+struct htu21d_data {
+	const struct device *i2c;
+	uint16_t temp;
```

Comment:
Temperature can't be negative? What's the range for this IC?

Reply:
-40 to 125°C